arelle/plugin/validate/NL/PluginValidationDataExtension.py

```diff
@@ -174,6 +175,31 @@ def getContextsWithSegments(self, modelXbrl: ModelXbrl) -> list[ModelContext | N
     def getFactLangFootnotes(self, modelXbrl: ModelXbrl) -> dict[ModelObject, set[str]]:
         return self.checkFootnotes(modelXbrl).factLangFootnotes
 
+    @lru_cache(1)
+    def getFilenameAllowedCharactersPattern(self):
+        return re.compile(
+            r"^[\w\.-]*$",
+            flags=re.ASCII
+        )
+
+    @lru_cache(1)
+    def getFilenameFormatPattern(self):
+        return re.compile(
+            r"^(?<base>\w*)-(?<date>[\w-]*)-(?<lang>\w*)\.(?<extension>html|htm|xhtml)$",
```

Comment:
Should this regex use `[^-]` rather than `\w`?  Otherwise, it seems like the code will produce a spurious NL-KVK.3.6.3.2 warning for a filename like `b@k-20250522-en.htm` that does follow the format (i.e., it seems like that filename should get NL-KVK.3.6.3.3 warning only).